includes/wikia/parser/templatetypes/tests/NavigationTemplateTest.php

```diff
@@ -58,4 +77,34 @@ public function getNavigationTemplates() {
 			],
 		];
 	}
+
+	/**
+	 * @dataProvider articleHtmlProvided
+	 */
+	public function testMultiTemplates( $marked, $expected, $message ) {
+		NavigationTemplate::resolve( $marked );
+
+		$this->assertEquals( $expected, $marked, $message );
+	}
+
+	public function articleHtmlProvided() {
+		return [
+			[ "", "", "Empty html should be correctly processed" ],
+			[
+				"\x7f26880747akjsdlkjflk <div>asdf</div>kasjdlfkjdks ksdjlafkj26880747\x7f26880747 aksdjlfkj alksjdldf\nlkjsdl \x7f26880747d26880747\x7f",
+				"26880747 aksdjlfkj alksjdldf\nlkjsdl d",
+				"If block element in navigation template it should be removed"
+			],
+			[
+				"\x7f26880747akjsdlkjflk <div>asdf</div>kasjdlfkjdks ksdjlafkj26880747\x7f test",
+				" test",
+				"Single nav template with block should be removed"
+			],
+			[
+				"\x7f26880747akjsdlkjflk <div>asdf</div>kasjdlfkjdks ksdjlafkj26880747\x7f test",
```

Comment:
Isn't it the duplicate of above case? Maybe you wanted to test for inline element and non-removal?